train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java

```diff
@@ -43,7 +43,9 @@ private void enforceSpeedLimit() {
 
 	@Override
 	public void setJoystickPosition(int joystickPosition) {
-		this.step = joystickPosition;		
+		this.step = joystickPosition;
+		this.followSpeed();
```

Comment:
The modification of the overrideJoysitckPosition function in TrainControllerImpl class affects the outcomes of UnitTests, as the followSpeed call is duplicated.
E.g: 
https://github.com/ftsrg-retelab/Sivatagi_rohamkukacok-vcs-ci-lab/blob/5fb94bdf86d9fcde9796961ae13ab945b28bf306/train-system/src/test/java/hu/bme/mit/train/system/TrainSystemTest.java#L46

Review UnitTests and remove unnecessary function calls.

Reply:
Done in commit 8b0e82b5c2d18d2d80bc576df47c894878a64db0